src/storage/txn/commands/acquire_pessimistic_lock.rs

```diff
@@ -87,6 +88,10 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for AcquirePessimisticLock
         let rows = keys.len();
         let mut res = if self.return_values {
             Ok(PessimisticLockRes::Values(vec![]))
+        } else if self.check_existence {
+            // If return_value is set, the existence status is implicitly included in the result.
+            // So check_existence only need to be explicitly handled if `return_values` is not set.
+            Ok(PessimisticLockRes::Existence(vec![]))
```

Comment:
Do we have cases to cover this new path for `AcquirePessimisticLock`?

Reply:
It should be covered by `test_pessimistic_lock` in storage/mod.rs. You can see if it's enough.

Reply:
And I'll split the pessimistic lock part to another PR.

Reply:
Got it, maybe we could add some explicit cases for it in another PR.